### ISSUE TITLE
pageInterval=day component onPageChange triggers always start of month

### ISSUE THREAD

**Issue by batuhansahan** [computerjazz/react-native-swipe-calendar#6]
Its always rendering same time which is start of month

also onDateSelect is not triggers how to know when the date change ?
```
      <Calendar   
        pageInterval="day"
        theme={{ todayIndicatorDotColor: 'blue', inactiveOpacity: 0.5 }}
        ref={calendarRef}           
        currentDate={currentDate}
        DayComponent={DayComponent}
        selectedDate={selectedDate}
        // currently showing 1 november if you go previous page like october than it shows 1 october etc.
        onPageChange={(date)=>alert(date.toDateString())} 
        monthAnimCallbackNode={monthAnimCallbackNode}
      /> 
```

tested in https://snack.expo.dev/@computerjazz/react-native-swipe-calendar

**Comment by tcolinpa** [comment]
@computerjazz hey, i'm having the same problem as @batuhansahan

same thing happens on pageInterval="month", the `onPageChange` returns the first day of the month, however it seems like the date just changes correctly when swiped nth the amount the weeks of the given month

```
 LOG  change 2022-11-01T04:04:09.328Z
 LOG  change 2022-12-01T04:04:09.328Z
 LOG  change 2022-12-01T04:04:09.328Z
 LOG  change 2022-12-01T04:04:09.328Z
 LOG  change 2022-12-01T04:04:09.328Z
 LOG  change 2022-12-01T04:04:09.328Z
 LOG  change 2023-01-01T04:04:09.328Z
 LOG  change 2023-01-01T04:04:09.328Z
 LOG  change 2023-01-01T04:04:09.328Z
 LOG  change 2023-01-01T04:04:09.328Z
 LOG  change 2023-02-01T04:04:09.328Z
 LOG  change 2023-02-01T04:04:09.328Z
 LOG  change 2023-02-01T04:04:09.328Z
 LOG  change 2023-02-01T04:04:09.328Z
 LOG  change 2023-03-01T04:04:09.328Z
 LOG  change 2023-03-01T04:04:09.328Z
 ```

**Comment by computerjazz** [comment]
Fixed in `0.0.21`
